### ISSUE TITLE
Invalid configuration files being generated by universal-cloudflared

### ISSUE THREAD

**Issue by TheCatLady** [linuxserver/docker-mods#280]
#256 introduced a bug that renders the `universal-cloudflared` plugin non-functional.

The last working version (which I have now pinned) appears to be `universal-cloudflared-2021.11.0-d62a1991b9e63a27d2809cccccb9f71f583b0349`.

### Logs
```
[cont-init.d] 98-cloudflared-config: executing...
**** Cloudflared setup script init... ****
**** Checking cloudflared setup script requirements... ****
**** Linux architecture found: amd64 ****
**** Linux distro found: alpine ****
**** Checking for cloudflared setup script dependencies... ****
**** Temporarily installing /tmp/yq... ****
**** jq already installed, skipping... ****
**** curl already installed, skipping... ****
**** Installing cloudflared...****
**** Moving /cloudflared/cloudflared-amd64 to /usr/local/bin/cloudflared... ****
**** Deleting tmp /cloudflared dir... ****
**** Cloudflared installed ****
cloudflared version 2022.1.0 (built 2022-01-06-1639 UTC)

**** Checking for optional cloudflare tunnel parameters... ****
**** Cloudflare tunnel parameters found, starting cloudflare tunnel setup... ****
**** Creating cloudflare tunnel(MY_REDACTED_TUNNEL_NAME) via API... ****
{
"message": "You already have a tunnel with this name. Delete the existing tunnel, or choose a different name for your new tunnel",
"code": 1013
}
jq: error (at <stdin>:1): Cannot iterate over null (null)

**** Saving cloudflare tunnel(MY_REDACTED_TUNNEL_NAME) credentials json... ****
null
**** Cloudflare tunnel(MY_REDACTED_TUNNEL_NAME) credentials saved to /etc/cloudflared/null.json ****
**** Generating config.yml for cloudflare tunnel(MY_REDACTED_TUNNEL_NAME)... ****
tunnel: null
credentials-file: /etc/cloudflared/null.json
no-autoupdate: true
ingress:
- service: https://localhost:443
originRequest:
originServerName: MY_REDACTED_SERVER_NAME
**** Config for cloudflare tunnel(MY_REDACTED_TUNNEL_NAME) saved to /etc/cloudflared/config.yml ****
**** Validating cloudflared tunnel rules... ****
Validating rules from /etc/cloudflared/config.yml
OK
**** Updating cloudflare zone... ****
**** Cleaning up cloudflared setup script dependencies if required... ****
**** Uninstalling /tmp/yq... ****
**** Cloudflared setup script done, exiting... ****
[cont-init.d] 98-cloudflared-config: exited 0.
```

**Comment by aptalca** [comment]
@pekovac @spunkie ^^

**Comment by pekovac** [comment]
@TheCatLady this seems to be the same issue why the PR was created for, API request with old version of the cloudflared mod would produce a 'null' response which would be saved into `/etc/cloudflared/null.json` due to change in CLoudflare API response, see #256. 

I could not create new tunnel at that point (#256), but in case there would be an existing tunnel at the time, this would be the response. 

Cloudlfared will at start query API for `id` of the installation, getting the `id` actually, but jq failing to parse the value ending up with `null.json` config.

Double checked with latest `lsio/nginx` image which is based off of Alpine 3.14 with the same version of `cloudflared version 2022.1.0 (built 2022-01-06-1639 UTC)` 

If your installation is docker-compose, could you please, try to `docker-compose down` and `docker-compose up` afterwards?

It seems that you are getting older version of this mod where `jq` scripts are still for the older version of the API.

Also, you will need to add `- service: http_status:404` to your ingress definition or else cloudflared will not start as per [documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file/ingress):
`The last rule you list in the config file must be a catch-all rule that matches all traffic.`

```yaml
ingress:
- service: https://localhost:443
```

**TLDR;**
I could not reproduce this issue with current version of the plugin, it seems plugin does not get updated during the container start-up, or fetches cached version. 

Checked on multiple VPS without any errors.

Docker compose:

```compose
version: '3'
services:
  nginx:
    image: linuxserver/nginx
    environment:
      PUID: 1000
      PGID: 1000
      DOCKER_MODS: linuxserver/mods:universal-cloudflared
      CF_ZONE_ID: <REDACTED>
      CF_ACCOUNT_ID: <REDACTED>
      CF_API_TOKEN: <REDACTED>
      CF_TUNNEL_NAME: nginx
      CF_TUNNEL_PASSWORD: <REDACTED>
      CF_TUNNEL_CONFIG: |
        ingress:
          - hostname: nginx.<REDACTED>
            service: hello_world
          - service: http_status:404
```

Log:

```
temp-nginx-1  | [cont-init.d] 98-cloudflared-config: executing... 
temp-nginx-1  | **** Cloudflared setup script init... ****
temp-nginx-1  | **** Checking cloudflared setup script requirements... ****
temp-nginx-1  | **** Linux architecture found: amd64 ****
temp-nginx-1  | **** Linux distro found: alpine ****
temp-nginx-1  | **** Checking for cloudflared setup script dependencies... ****
temp-nginx-1  | **** Temporarily installing /tmp/yq... ****
temp-nginx-1  | **** jq already installed, skipping... ****
temp-nginx-1  | **** curl already installed, skipping... ****
temp-nginx-1  | **** Installing cloudflared...****
temp-nginx-1  | **** Cloudflared already installed, skipping... ****
temp-nginx-1  | cloudflared version 2022.1.0 (built 2022-01-06-1639 UTC)
temp-nginx-1  | **** Checking for optional cloudflare tunnel parameters... ****
temp-nginx-1  | **** Cloudflare tunnel parameters found, starting cloudflare tunnel setup... ****
temp-nginx-1  | **** Creating cloudflare tunnel(nginx) via API... ****
temp-nginx-1  | {
temp-nginx-1  |   "success": false,
temp-nginx-1  |   "errors": [
temp-nginx-1  |     {
temp-nginx-1  |       "code": 1013,
temp-nginx-1  |       "message": "You already have a tunnel with this name. Delete the existing tunnel, or choose a different name for your new tunnel"
temp-nginx-1  |     }
temp-nginx-1  |   ],
temp-nginx-1  |   "messages": [],
temp-nginx-1  |   "result": null
temp-nginx-1  | }
temp-nginx-1  | **** You already have a cloudflare tunnel named nginx ****
temp-nginx-1  | **** Searching existing cloudflare tunnels via API... ****
temp-nginx-1  | {
temp-nginx-1  |   "success": true,
temp-nginx-1  |   "messages": [],
temp-nginx-1  |   "errors": [],
temp-nginx-1  |   "result": [
temp-nginx-1  |     {
temp-nginx-1  |       "id": "<REDACTED>",
temp-nginx-1  |       "account_tag": "<REDACTED>",
temp-nginx-1  |       "created_at": "2022-01-08T11:08:30.560203Z",
temp-nginx-1  |       "deleted_at": null,
temp-nginx-1  |       "name": "nginx",
temp-nginx-1  |       "connections": [],
temp-nginx-1  |       "conns_active_at": null,
temp-nginx-1  |       "conns_inactive_at": "2022-01-08T11:09:00.500264Z",
temp-nginx-1  |       "tun_type": "cfd_tunnel",
temp-nginx-1  |       "metadata": {},
temp-nginx-1  |       "status": "inactive"
temp-nginx-1  |     }
temp-nginx-1  |   ]
temp-nginx-1  | }
temp-nginx-1  | **** Fetching existing cloudflare tunnel(nginx) via API... ****
temp-nginx-1  | {
temp-nginx-1  |   "success": true,
temp-nginx-1  |   "messages": [],
temp-nginx-1  |   "errors": [],
temp-nginx-1  |   "result": {
temp-nginx-1  |     "id": "<REDACTED>",
temp-nginx-1  |     "account_tag": "<REDACTED>",
temp-nginx-1  |     "created_at": "2022-01-08T11:08:30.560203Z",
temp-nginx-1  |     "deleted_at": null,
temp-nginx-1  |     "name": "nginx",
temp-nginx-1  |     "connections": [],
temp-nginx-1  |     "conns_active_at": null,
temp-nginx-1  |     "conns_inactive_at": "2022-01-08T11:09:00.500264Z",
temp-nginx-1  |     "tun_type": "cfd_tunnel",
temp-nginx-1  |     "metadata": {},
temp-nginx-1  |     "status": "inactive"
temp-nginx-1  |   },
temp-nginx-1  |   "credentials_file": {
temp-nginx-1  |     "AccountTag": "<REDACTED>",
temp-nginx-1  |     "TunnelID": "<REDACTED>",
temp-nginx-1  |     "TunnelName": "nginx",
temp-nginx-1  |     "TunnelSecret": "<REDACTED>"
temp-nginx-1  |   }
temp-nginx-1  | }
temp-nginx-1  | **** Saving cloudflare tunnel(nginx) credentials json... ****
temp-nginx-1  | null
temp-nginx-1  | **** Cloudflare tunnel(nginx) credentials saved to /etc/cloudflared/<REDACTED>.json ****
temp-nginx-1  | **** Generating config.yml for cloudflare tunnel(nginx)... ****
temp-nginx-1  | tunnel: <REDACTED>
temp-nginx-1  | credentials-file: /etc/cloudflared/<REDACTED>.json
temp-nginx-1  | no-autoupdate: true
temp-nginx-1  | ingress:
temp-nginx-1  |   - hostname: nginx.<REDACTED>
temp-nginx-1  |     service: hello_world
temp-nginx-1  |   - service: http_status:404
temp-nginx-1  | **** Config for cloudflare tunnel(nginx) saved to /etc/cloudflared/config.yml ****
temp-nginx-1  | **** Validating cloudflared tunnel rules... ****
temp-nginx-1  | Validating rules from /etc/cloudflared/config.yml
temp-nginx-1  | OK
temp-nginx-1  | **** Updating cloudflare zone... ****
temp-nginx-1  | **** Searching zone for hostname(nginx.<REDACTED>) via API... ****
temp-nginx-1  | **** Updating existing CNAME for hostname(nginx.<REDACTED>) via API... ****
temp-nginx-1  | {
temp-nginx-1  |   "result": {
temp-nginx-1  |     "id": "<REDACTED>",
temp-nginx-1  |     "zone_id": "<REDACTED>",
temp-nginx-1  |     "zone_name": "<REDACTED>",
temp-nginx-1  |     "name": "nginx.<REDACTED>",
temp-nginx-1  |     "type": "CNAME",
temp-nginx-1  |     "content": "<REDACTED>.cfargotunnel.com",
temp-nginx-1  |     "proxiable": true,
temp-nginx-1  |     "proxied": true,
temp-nginx-1  |     "ttl": 1,
temp-nginx-1  |     "locked": false,
temp-nginx-1  |     "meta": {
temp-nginx-1  |       "auto_added": false,
temp-nginx-1  |       "managed_by_apps": false,
temp-nginx-1  |       "managed_by_argo_tunnel": false
temp-nginx-1  |     },
temp-nginx-1  |     "created_on": "2022-01-08T11:06:07.476026Z",
temp-nginx-1  |     "modified_on": "2022-01-08T11:08:33.290704Z"
temp-nginx-1  |   },
temp-nginx-1  |   "success": true,
temp-nginx-1  |   "errors": [],
temp-nginx-1  |   "messages": []
temp-nginx-1  | }
temp-nginx-1  | **** Cleaning up cloudflared setup script dependencies if required... ****
temp-nginx-1  | **** Uninstalling /tmp/yq... ****
temp-nginx-1  | **** Cloudflared setup script done, exiting... ****
temp-nginx-1  | [cont-init.d] 98-cloudflared-config: exited 0.
temp-nginx-1  | [cont-init.d] 99-custom-files: executing... 
temp-nginx-1  | [custom-init] no custom files found exiting...
temp-nginx-1  | [cont-init.d] 99-custom-files: exited 0.
temp-nginx-1  | [cont-init.d] done.
temp-nginx-1  | [services.d] starting services
temp-nginx-1  | [services.d] done.
temp-nginx-1  | 2022-01-08T11:09:13Z INF Starting tunnel tunnelID=<REDACTED>
temp-nginx-1  | 2022-01-08T11:09:13Z INF Version 2022.1.0
temp-nginx-1  | 2022-01-08T11:09:13Z INF GOOS: linux, GOVersion: go1.17.5, GoArch: amd64
temp-nginx-1  | 2022-01-08T11:09:13Z INF Settings: map[config:/etc/cloudflared/config.yml cred-file:/etc/cloudflared/<REDACTED>.json credentials-file:/etc/cloudflared/<REDACTED>.json no-autoupdate:true]
temp-nginx-1  | 2022-01-08T11:09:13Z INF Environmental variables map[CF_TUNNEL_CONFIG:ingress:
temp-nginx-1  |   - hostname: nginx.<REDACTED>
temp-nginx-1  |     service: hello_world
temp-nginx-1  |   - service: http_status:404 CF_TUNNEL_NAME:nginx CF_TUNNEL_PASSWORD:<REDACTED>#optional]
temp-nginx-1  | 2022-01-08T11:09:13Z INF Generated Connector ID: <REDACTED>
temp-nginx-1  | 2022-01-08T11:09:13Z INF Initial protocol http2
temp-nginx-1  | 2022-01-08T11:09:13Z INF Starting metrics server on 127.0.0.1:40347/metrics
temp-nginx-1  | 2022-01-08T11:09:14Z INF Starting Hello World server at 127.0.0.1:41201
temp-nginx-1  | 2022-01-08T11:09:14Z INF Connection 2870c7ff-276e-4ae0-9ca7-84f181b0ed52 registered connIndex=0 location=VIE
temp-nginx-1  | 2022-01-08T11:09:15Z INF Connection 6121a0e6-a28d-48d0-b1b4-ef9b5c46d729 registered connIndex=1 location=PRG
temp-nginx-1  | 2022-01-08T11:09:15Z INF Connection e48ea7be-b01c-4b7b-93f8-324c24cf6425 registered connIndex=2 location=VIE
temp-nginx-1  | 2022-01-08T11:09:17Z INF Connection 895d33e5-d25e-43a2-8098-d878f169c45d registered connIndex=3 location=PRG
```

**Comment by TheCatLady** [comment]
> It seems that you are getting older version of this mod where `jq` scripts are still for the older version of the API.

The issue is that I _am_ getting the newer versions of the mod.  The last working version (noted above) was the one built prior to your PR being merged.

> Also, you will need to add `- service: http_status:404` to your ingress definition or else cloudflared will not start as per [documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file/ingress):
> `The last rule you list in the config file must be a catch-all rule that matches all traffic.`
> 
> ```yaml
> ingress:
> - service: https://localhost:443
> ```

I have a single rule that matches all traffic, and `cloudflared` is starting without issue as long as I use an older version of the mod.

**Comment by pekovac** [comment]
@TheCatLady 

I did check the [API definition](https://api.cloudflare.com/#argo-tunnel-create-argo-tunnel) on `cloudflared` regarding creating Argo Tunnels and double checked the error response in your log and what I'm getting.

Your log:
```log
**** Creating cloudflare tunnel(MY_REDACTED_TUNNEL_NAME) via API... ****
{
"message": "You already have a tunnel with this name. Delete the existing tunnel, or choose a different name for your new tunnel",
"code": 1013
}
jq: error (at <stdin>:1): Cannot iterate over null (null)
```

My log:
```log
 **** Creating cloudflare tunnel(nginx) via API... ****
 {
   "success": false,
   "errors": [
     {
       "code": 1013,
       "message": "You already have a tunnel with this name. Delete the existing tunnel, or choose a different name for your new tunnel"
     }
   ],
   "messages": [],
   "result": null
 }
```

You are getting different response as it should be, most probably cached upstream or hitting a server that was not updated yet, or there is a caching proxy in-between, that's why pinning the version works for you.

Old code checks json response for `jq -rc ".code"` while new one checks for `jq -rc ".errors[].code"`. With the response you are getting it fails on `null` as it cannot find proper key.

This means that code is working properly, problem is with API response.

**Comment by TheCatLady** [comment]
@pekovac 

> You are getting different response as it should be, most probably cached upstream or hitting a server that was not updated yet, or there is a caching proxy in-between, that's why pinning the version works for you.
> 
> Old code checks json response for `jq -rc ".code"` while new one checks for `jq -rc ".errors[].code"`. With the response you are getting it fails on `null` as it cannot find proper key.
> 
> This means that code is working properly, problem is with API response.

Respectfully, this does _not_ mean that the code is working properly. 😅  The code should handle both types of responses gracefully.

Also, I don't know if this is a caching issue.  Your PR was merged over two months ago.  We should just assume that it's possible for either type of response to be returned.

**Comment by pekovac** [comment]
@TheCatLady 

I have no reference for the old API definitions so I cannot test it. 
Please, feel free to open a PR with the changes you require.

**Comment by TheCatLady** [comment]
Fixed in #281
